### PR TITLE
Case 21271: Remove unnessary picks for quest

### PR DESCRIPTION
**Pull request by danteruiz**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17326/HighFidelity-Beta-PR14933-13da2f9.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17326/HighFidelity-Beta-Interface-PR14933-13da2f9.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17326/HighFidelity-Beta-PR14933-13da2f9.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17326/HighFidelity-Beta-Interface-PR14933-13da2f9.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17353/HighFidelity-Beta-PR14933-b00746c.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17353/HighFidelity-Beta-PR14933-b00746c.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17353/HighFidelity-Beta-Interface-PR14933-b00746c.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17353/HighFidelity-Beta-PR14933-b00746c.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17353/HighFidelity-Beta-Interface-PR14933-b00746c.exe

File: interface/src/raypick/PickScriptingInterface.cpp [hunk 1]
```diff
@@ -71,6 +71,18 @@ PickFilter getPickFilter(unsigned int filter) {
 unsigned int PickScriptingInterface::createRayPick(const QVariant& properties) {
     QVariantMap propMap = properties.toMap();
 
+
+#if defined (Q_OS_ANDROID)
+    QString jointName { "" };
+    if (propMap["joint"].isValid()) {
+        QString jointName = propMap["joint"].toString();
+        QString mouseJoint { "Mouse" };
```
Comment (SamGondelman) [inline interface/src/raypick/PickScriptingInterface.cpp]:
```suggestion
        const QString MOUSE = "Mouse";
```
(same elsewhere)

File: libraries/pointers/src/PickCacheOptimizer.h [hunk 2]
```diff
@@ -11,6 +11,8 @@
 #include <unordered_map>
 
 #include "Pick.h"
+#include "PerfStat.h"
+#include "Profile.h"
```
Comment (SamGondelman) [inline libraries/pointers/src/PickCacheOptimizer.h]: needed?

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17394/HighFidelity-Beta-PR14933-4bbae12.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17394/HighFidelity-Beta-PR14933-4bbae12.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17394/HighFidelity-Beta-Interface-PR14933-4bbae12.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17394/HighFidelity-Beta-PR14933-4bbae12.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17394/HighFidelity-Beta-Interface-PR14933-4bbae12.exe

**Comment by danteruiz** [comment]
Test plan 
1. launch interface and open up stats and expand it, there should be 19 total raypicks and 9 Intersection calls for rays on desktop.  
2. confirm that general controller script functionally behaves the same way

**Pull request merged**